### ISSUE TITLE
Use `DUK_USE_DATE_NOW_WINDOWS` flag for Windows 7 support

### ISSUE THREAD

**Issue by efoerster** [SkylerLipthay/ducc#3]
The library currently only works on Windows 7 if it was compiled on a Windows 7 (or older) machine. See this code from `duk_config_default.h`:

```
/* GetSystemTimePreciseAsFileTime() available from Windows 8:
 * https://msdn.microsoft.com/en-us/library/windows/desktop/hh706895(v=vs.85).aspx
 */
#if defined(DUK_USE_DATE_NOW_WINDOWS_SUBMS) || defined(DUK_USE_DATE_NOW_WINDOWS)
/* User forced provider. */
#else
#if defined(_WIN32_WINNT) && (_WIN32_WINNT >= 0x0602)
#define DUK_USE_DATE_NOW_WINDOWS_SUBMS
#else
#define DUK_USE_DATE_NOW_WINDOWS
#endif
#endif
```
We use this library and ran into this issue, see https://github.com/latex-lsp/texlab/issues/101. Therefore, it would be nice to have a feature flag for `ducc` to enable Windows 7 support.

**Comment by SkylerLipthay** [comment]
Hm, I see... So `DUK_USE_DATE_NOW_WINDOWS_SUBMS` is being defined during compilation on Windows 8/10 but then `GetSystemTimePreciseAsFileTime` fails to resolve at runtime under Windows 7. That's an unfortunate consequence. If I were on the Duktape team I would have done it the other way around. (So that the user would specify a flag to *activate* this poorly supported sub-millisecond accuracy.)

I'm tempted to just `#define DUK_USE_DATE_NOW_WINDOWS`. If I'm understanding things correctly, this would sacrifice sub-millisecond timer accuracy on Windows 8/10, but at least it would fix this issue. I highly doubt this would affect any current ducc users... Besides, supporting such granular feature flags seems against the nature of ducc, which intentionally hides such implementation details.

Is this OK with you?

(Alternatively it looks like you can supply your own time functions that could perhaps dynamically fallback to `GetSystemTime` if `GetSystemTimePreciseAsFileTime` is missing. I'm guessing this is overkill for your purposes.)

**Comment by efoerster** [comment]
Sure, thats even better. Thank you!

**Comment by SkylerLipthay** [comment]
Alright give `ducc 0.1.1` a shot. Close this issue if that fixes the user's issue or let me know if further work is needed :)

**Comment by SkylerLipthay** [comment]
Apologies @efoerster. It just dawned on me that `0.1.1` probably failed to fix the problem. I misunderstood the logic required of `duk_config.h`. I have limited testing capabilities with Windows... but it should be fixed now (302d7a4) with `0.1.2`.

**Comment by efoerster** [comment]
Seems to be fixed, thanks for your fast response.
